### PR TITLE
add Dockerfile for local development without python

### PR DESCRIPTION
**Pull request by lpmi-13**
- Add a Dockerfile to run the application without needing to install any python dependencies (or even python itself!)
- Update app.py to bind to 0.0.0.0 in DEV mode so it can respond inside a container
- Update Contributing.md with instructions for building and running a containerized version of the webapp

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+.git
+.github
+*.md
+.dockerignore
```
Comment (lpmi-13) [inline .dockerignore:4]: I will forever lol at the dockerignore needing to ignore itself

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+FROM ubuntu:20.04
```
Comment (lpmi-13) [inline Dockerfile:1]: I expected this to be a lot heavier, but the final image size, even with the full application, is only 485MB. I'm open to trying to slim that down if it's still too big.

**Comment by fpereiro** [comment]
Hi @lpmi-13 ! Thanks for this - I'll review early next week!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -56,6 +56,15 @@ If you want to run the website version locally, run:
 (.env)$ python app.py
 ```
 
+If you want to run the website locally, but would prefer to use Docker instead of installing python, you can build a container and run it like so:
+```bash
+docker build -t NAME_OF_CONTAINER .
```
Comment (tiborvass) [inline CONTRIBUTING.md:61]: technically it's the name of the image not the name of the container. I would instead just name the image `hedy` instead.

File: .dockerignore (hunk 1 above)
Comment (tiborvass) [reply .dockerignore:4]: I agree it's a bit weird. It was modeled after gitignore which does have different design considerations but oh well.

File: Dockerfile (hunk 2 above)
Comment (tiborvass) [reply Dockerfile:1]: You can reduce the image to 200MB by remove the `apt install` and doing `FROM python:3.8-slim` (might as well update to 3.9 or 3.10), as well as using `pip3 install --no-cache-dir -r /tmp/requirements.txt`

**Comment by fpereiro** [comment]
Hi @lpmi-13 ! Haven't forgotten about this! We're hard at work finishing https://github.com/Felienne/hedy/pull/299; when that's cleared out I will take a look at this. I'll keep you updated.

**Comment by lpmi-13** [comment]
No rush at all. I'll update this PR with @tiborvass excellent suggestions in the meantime!

File: Dockerfile (hunk 2 above)
Comment (lpmi-13) [reply Dockerfile:1]: couldn't find any 3.10 slim images that had gcc, so going with `3.9-slim`, which "just works" (:tm: )

File: CONTRIBUTING.md (hunk 3 above)
Comment (lpmi-13) [reply CONTRIBUTING.md:61]: this is much simpler, and very sensible. thanks!

**Comment by lpmi-13** [comment]
Have updated based on feedback, and do please let me know if you would like me to rebase at some point or if one of the maintainers would prefer to.

**Review by fpereiro — APPROVED** [review]

**Comment by fpereiro** [comment]
Hi @lpmi-13 ! I finally got around to testing it. It works brilliantly!

There's a small conflict in CONTRIBUTING.md - if you can fix that (by adding both texts, yours and the updated version from master), we'll be good to merge.

**Comment by lpmi-13** [comment]
ah darnit...I tried to be too clever and fix it without needing to rebase, but I couldn't figure out how to do it, so I'm just gonna rebase and then update.

**Comment by lpmi-13** [comment]
alright, should be all good now!

**Review by fpereiro — APPROVED** [review]

**Comment by fpereiro** [comment]
@lpmi-13 merged! Thank you again for this!

**Pull request merged**